src/ucs/datastruct/usage_tracker.c

```diff
@@ -14,51 +14,48 @@
 #include <ucs/debug/memtrack_int.h>
 
 
-ucs_status_t
-ucs_usage_tracker_create(unsigned ticks_per_flush, unsigned active_capacity,
-                         double active_thresh, unsigned eject_thresh,
-                         ucs_usage_tracker_flush_cb_t cb, void *opaque,
-                         ucs_usage_tracker_h *usage_tracker_p)
+ucs_status_t ucs_usage_tracker_create(const ucs_usage_tracker_params_t *params,
+                                      ucs_usage_tracker_h *usage_tracker_p)
 {
     ucs_status_t status;
     ucs_usage_tracker_h usage_tracker;
 
+    if (params->flush_cb == NULL) {
+        status = UCS_ERR_INVALID_PARAM;
+        goto err;
+    }
+
     usage_tracker = ucs_calloc(1, sizeof(*usage_tracker), "ucs_usage_tracker");
     if (usage_tracker == NULL) {
-        ucs_error("failed to allocate Usage Tracker (ticks_per_flush %u, "
-                  "capacity: %u, "
-                  "active_thresh: %.2f, eject_thresh %u)",
-                  ticks_per_flush, active_capacity, active_thresh,
-                  eject_thresh);
+        ucs_error("failed to allocate usage tracker");
         status = UCS_ERR_NO_MEMORY;
         goto err;
     }
 
-    status = ucs_lru_create(active_capacity, &usage_tracker->lru);
+    status = ucs_lru_create(params->active_capacity, &usage_tracker->lru);
     if (status != UCS_OK) {
         goto err_free_tracker;
     }
 
-    kh_init_inplace(usage_tracker_hash, &usage_tracker->hash);
+    kh_init_no_shrink(usage_tracker_hash, &usage_tracker->hash);

```

Comment:
why needed?

Reply:
to avoid shrinking, which reallocate all the khash mem and corrupts the LRU linked list.

Reply:
we need to use ucs_hlist as hash element and not regular list
when hash size increase it also reallocates memory